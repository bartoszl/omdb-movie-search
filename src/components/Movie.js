import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const MovieContainer = styled.div`
  margin: 4px 8px 16px 8px;
  width: 248px;
  padding: 4px;
  border-radius: 3px;
  background-color: black;
  color: white;
  box-shadow: 0 0 5px 1px #aaa;
`;

const MoviePoster = styled.img`
  width: 100%;
  height: auto;
`;

const Movie = ({ imgSrc, title }) => (
  <MovieContainer>
    <MoviePoster src={imgSrc} alt={title} />
    <h4>
      { title }
    </h4>
  </MovieContainer>
);

Movie.propTypes = {
  imgSrc: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default Movie;
