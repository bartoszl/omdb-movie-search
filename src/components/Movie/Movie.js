import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { BACKGROUND_GREY } from '../../constants/colors';

const MovieContainer = styled(Link)`
  margin: 4px 8px 16px 8px;
  width: 248px;
  padding: 4px;
  border-radius: 3px;
  background-color: ${BACKGROUND_GREY};
  color: white;
  box-shadow: ${({ theme }) => theme.shadow.small};
  transition: all 0.25s;
  text-decoration: none;

  display: flex;
  flex-direction: column;
  justify-content: space-between;

  &:hover {
    box-shadow: ${({ theme }) => theme.shadow.large};
    padding: 8px;
    margin: 0px 4px 12px 4px;
  }
`;

const MoviePoster = styled.img`
  width: 100%;
  height: auto;
`;

const Title = styled.h4`
  margin-top: 0.75rem;
  margin-bottom: 0.25rem;
`;

const Year = styled.h5`
  margin-top: 0.25rem;
  margin-bottom: 0.5rem;
`;

const Movie = ({
  imgSrc, title, year, to,
}) => (
  <MovieContainer to={to}>
    <MoviePoster src={imgSrc} alt={title} />
    <div>
      <Title>
        { title }
      </Title>
      <Year>
        { year }
      </Year>
    </div>
  </MovieContainer>
);

Movie.propTypes = {
  imgSrc: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  year: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
};

export default Movie;
