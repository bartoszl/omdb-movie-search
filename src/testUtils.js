import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import theme from './config/theme';

const TestProviders = ({ children }) => (
  <ThemeProvider theme={theme}>
    <MemoryRouter>
      {children}
    </MemoryRouter>
  </ThemeProvider>
);

TestProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) => render(ui, { wrapper: TestProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
