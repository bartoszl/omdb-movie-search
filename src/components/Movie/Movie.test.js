import React from 'react';
import { render, cleanup, screen } from '../../testUtils';
import Movie from './Movie';

describe('InfoLabel', () => {
  const title = 'Title';
  const imgSrc = 'www.image.pl';
  const year = '1993';

  beforeEach(() => {
    render((
      <Movie
        title={title}
        imgSrc={imgSrc}
        year={year}
        to="asd"
      />
    ));
  });

  afterEach(() => {
    cleanup();
  });

  it('renders title', () => {
    expect(screen.getByText(title)).toBeTruthy();
  });

  it('renders year', () => {
    expect(screen.getByText(year)).toBeTruthy();
  });

  it('renders poster', () => {
    expect(screen.getByAltText(title)).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { asFragment } = render((
      <Movie
        title={title}
        imgSrc={imgSrc}
        year={year}
        to="asd"
      />
    ));

    expect(asFragment()).toMatchSnapshot();
  });
});
