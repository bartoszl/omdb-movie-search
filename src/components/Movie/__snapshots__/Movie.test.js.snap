// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoLabel matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="sc-bdnylx dwuzaO"
    href="/asd"
  >
    <img
      alt="Title"
      class="sc-gtssRu ePsuqd"
      src="www.image.pl"
    />
    <div>
      <h4
        class="sc-dlnjPT jAnuC"
      >
        Title
      </h4>
      <h5
        class="sc-hKFyIo gaWfhD"
      >
        1993
      </h5>
    </div>
  </a>
</DocumentFragment>
`;
